src/routes/(app)/product/[id]/+page.server.ts

```diff
@@ -145,13 +153,17 @@ async function addToCart({ params, request, locals }: RequestEvent) {
 				.regex(/^\d+(\.\d+)?$/)
 				.optional(),
 			customPriceCurrency: z.enum([CURRENCIES[0], ...CURRENCIES.slice(1)]).optional(),
-			deposit: z.enum(['partial', 'full']).optional()
+			deposit: z.enum(['partial', 'full']).optional(),
+			chosenVariations: z.record(z.string()).optional()
 		})
 		.parse({
 			quantity: formData.get('quantity') || undefined,
 			customPriceAmount: formData.get('customPriceAmount') || undefined,
 			customPriceCurrency: formData.get('customPriceCurrency') || undefined,
-			deposit: formData.get('deposit') || undefined
+			deposit: formData.get('deposit') || undefined,
+			chosenVariations: formData.get('chosenVariations')
+				? JSON.parse(formData.get('chosenVariations')?.toString() || '[]')
+				: undefined
```

Comment:
Use this if you must:

```ts
		const json: JsonObject = {};
		for (const [key, value] of await request.formData()) {
			if (value) {
				set(json, key, value);
			}
		}
```

but no `JSON.parse` here please, it can cause 500 errors